src/thread.rs

```diff
@@ -248,6 +248,12 @@ impl<'mir, 'tcx: 'mir> ThreadManager<'mir, 'tcx> {
         self.threads[thread_id].state == ThreadState::Terminated
     }
 
+    /// Enable the thread for execution. The thread must be terminated.
+    fn enable_thread(&mut self, thread_id: ThreadId) {
```

Comment:
This seems rather prone to misuse... maybe we should have another state, `ThreadState::TlsDtors` or so?

Reply:
Hm I see it actually basically goes back to normal execution during dtor running, which makes sense. So this seems fine.